### PR TITLE
Create Invite API and lockable skeleton

### LINKED ISSUE #37: Create "resend invite" API for Employees

### PR DESCRIPTION
**Pull request by iv22**

### CODE REVIEW

File: app/policies/employees/credentials.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Employees
+  module Credentials
```
Comment (anatoliliotych) [inline app/policies/employees/credentials.rb]: confusing naming

File: app/services/employees/invitation_assistant.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  class InvitationAssistant < ApplicationService
+    attr_reader :member
+
+    def initialize(member)
+      super()
+      @member = member
+    end
+
+    def call
+      if member.status == 'active'
```
Comment (anatoliliotych) [inline app/services/employees/invitation_assistant.rb]: too much conditions

Comment (anatoliliotych) [reply app/services/employees/invitation_assistant.rb]: refactor this code

File: spec/factories/users.rb [hunk 3]
```diff
@@ -60,6 +62,10 @@
       role { 2 }
     end
 
+    trait :locked do
+      locked_at { Time.now }
```
Comment (anatoliliotych) [inline spec/factories/users.rb]: this can play bad joke, use Time.zone.now

File: app/policies/employees/user_policy.rb [hunk 4]
```diff
@@ -3,6 +3,8 @@
 module Employees
   # Check current User credentials
   class UserPolicy < ApplicationPolicy
+    include Credentials
```
Comment (anatoliliotych) [inline app/policies/employees/user_policy.rb]: why we use this? as all you've moved is related to user

File: app/controllers/api/v1/employees_controller.rb [hunk 5]
```diff
@@ -3,16 +3,10 @@
...
   module V1
     # Employees API controller
-    class EmployeesController < ApplicationController
-      before_action :authenticate_user!
-      rescue_from ActiveRecord::RecordNotFound, with: :record_not_found
-      rescue_from ActiveRecord::RecordInvalid, with: :record_invalid
-      rescue_from ActionPolicy::Unauthorized, with: :handle_unauthorized
-      rescue_from ArgumentError, with: :params_invalid
-
+    class EmployeesController < UserController
```
Comment (anatoliliotych) [inline app/controllers/api/v1/employees_controller.rb]: I don't think that it's a good idea to inherit from contoller, just use modules if needed

**Review by anatoliliotych — CHANGES_REQUESTED** [review]

File: app/policies/employees/user_policy.rb (hunk 4 above)
Comment (iv22) [reply app/policies/employees/user_policy.rb]: the included module provide permission checker
this module going to use in other policies like invitation, locking, teams...
I propose to rename Credentials module to PermissionChecker

File: app/controllers/api/v1/employees_controller.rb (hunk 5 above)
Comment (iv22) [reply app/controllers/api/v1/employees_controller.rb]: my aim was following DRY principle
I don't want repeat this code in each controller:
```
      before_action :authenticate_user!
      rescue_from ActiveRecord::RecordNotFound, with: :record_not_found
      rescue_from ActiveRecord::RecordInvalid, with: :record_invalid
      rescue_from ActionPolicy::Unauthorized, with: :handle_unauthorized
      rescue_from ArgumentError, with: :params_invalid
```
I have seen this method few times, for example: https://rmulhol.github.io/ruby/2015/04/27/solid-rails.html

File: app/policies/employees/user_policy.rb (hunk 4 above)
Comment (anatoliliotych) [reply app/policies/employees/user_policy.rb]: let's rename

File: app/controllers/api/v1/employees_controller.rb (hunk 5 above)
Comment (anatoliliotych) [reply app/controllers/api/v1/employees_controller.rb]: yeah, but you inherit from UserController, use Application instead

File: app/controllers/api/v1/employee_utils.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module EmployeeUtils
```
Comment (anatoliliotych) [inline app/controllers/api/v1/employee_utils.rb]: put it in concerns

File: app/controllers/api/v1/employees_controller.rb [hunk 7]
```diff
@@ -23,14 +21,17 @@ def show
 
       def create
         authorize! Object, with: Employees::UserPolicy
-        invited_member = Employees::MemberCreator.call(member_params, company)
-        render json: invited_member, status: :created, location: api_v1_employee_path(invited_member)
+        invited_member = Employees::MemberCreator.call(member_params, company).decorate
+        render json: invited_member,
```
Comment (anatoliliotych) [inline app/controllers/api/v1/employees_controller.rb]: if you use this style, use it everywhere...

File: spec/services/employees/invitation_assistant_spec.rb [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    end
+  end
+
+  context 'when try to resend invite for invited employee' do
+    before do
+      member = FactoryBot.create(:employee_user, :invited).decorate
+      @response = Employees::InvitationAssistant.call(member)
+    end
+
+    it "should accept operation" do
```
Comment (anatoliliotych) [inline spec/services/employees/invitation_assistant_spec.rb]: single quotes please everywhere

**Review by anatoliliotych — CHANGES_REQUESTED** [review]

**Review by anatoliliotych — APPROVED** [review]

**Pull request merged**